Annotation reply is null

### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.6.16

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

step 1. Enable the annotated reply
step 2. Reply is empty in frondend if the answer hits the annotated content, and an error occurs in backend

### ✔️ Expected Behavior

Reply the annotated content

### ❌ Actual Behavior

```python
[2024-08-07 01:56:33 +0000] [46] [ERROR] Error handling request /api/chat-messages
Traceback (most recent call last):
  File "/app/api/.venv/lib/python3.10/site-packages/gunicorn/workers/base_async.py", line 55, in handle
    self.handle_request(listener_name, req, client, addr)
  File "/app/api/.venv/lib/python3.10/site-packages/gunicorn/workers/ggevent.py", line 128, in handle_request
    super().handle_request(listener_name, req, sock, addr)
  File "/app/api/.venv/lib/python3.10/site-packages/gunicorn/workers/base_async.py", line 115, in handle_request
    for item in respiter:
  File "/app/api/.venv/lib/python3.10/site-packages/werkzeug/wsgi.py", line 256, in __next__
    return self._next()
  File "/app/api/.venv/lib/python3.10/site-packages/werkzeug/wrappers/response.py", line 32, in _iter_encoded
    for item in iterable:
  File "/app/api/.venv/lib/python3.10/site-packages/flask/helpers.py", line 113, in generator
    yield from gen
  File "/app/api/libs/helper.py", line 168, in generate
    yield from response
  File "/app/api/core/app/features/rate_limiting/rate_limit.py", line 110, in __next__
    return next(self.generator)
  File "/app/api/core/app/apps/base_app_generate_response_converter.py", line 37, in _generate_simple_response
    for chunk in cls.convert_stream_simple_response(response):
  File "/app/api/core/app/apps/advanced_chat/generate_response_converter.py", line 94, in convert_stream_simple_response
    for chunk in stream_response:
  File "/app/api/core/app/apps/advanced_chat/generate_task_pipeline.py", line 179, in _to_stream_response
    for stream_response in generator:
  File "/app/api/core/app/apps/advanced_chat/generate_task_pipeline.py", line 206, in _wrapper_process_stream_response
    for response in self._process_stream_response(publisher=publisher, trace_manager=trace_manager):
  File "/app/api/core/app/apps/advanced_chat/generate_task_pipeline.py", line 247, in _process_stream_response
    if hasattr(message.event, 'metadata') and message.event.metadata.get('is_answer_previous_node', False) and publisher:
AttributeError: 'NoneType' object has no attribute 'get'
```

--- a/api/core/app/apps/advanced_chat/generate_task_pipeline.py
+++ b/api/core/app/apps/advanced_chat/generate_task_pipeline.py
@@ -244,7 +244,11 @@ class AdvancedChatAppGenerateTaskPipeline(BasedGenerateTaskPipeline, WorkflowCyc
         :return:
         """
         for message in self._queue_manager.listen():
-            if message.event and hasattr(message.event, 'metadata') and message.event.metadata.get('is_answer_previous_node', False) and publisher:
+            if (message.event
+                    and hasattr(message.event, 'metadata')
+                    and message.event.metadata
+                    and message.event.metadata.get('is_answer_previous_node', False)
+                    and publisher):
                 publisher.publish(message=message)
             elif (hasattr(message.event, 'execution_metadata')
                   and message.event.execution_metadata

